### DISCUSSION TITLE
"Go to definition" doesn't work for token has union of rules

### DISCUSSION THREAD

**Discussion by jindong-zhannng** [eclipse-langium/langium#1548] [Q&A] [answered]
Hi there,

I'm implementing a TS-like language including type alias:

```ts
type MyType = something
              // ^ can be a built-in type, inline type definition or a type reference
```

To achieve it, here is partial of my grammar:

```
TypeAliasDeclaration:
  'type' name=ID '=' type=TypeExpr;

TypeExpr:
  PrimitiveTypeExpr | CompositeTypeExpr | TypeRefExpr;

TypeRefExpr:
  ref=[TypeDeclaration];

TypeDeclaration:
  TypeAliasDeclaration | TypeParameterDeclaration;

TypeParameterDeclaration:
  name=ID;
```

I expect that `TypeExpr` can be a cross-reference to `TypeAliasDeclaration` or `TypeParameterDeclaration`. It works on parsing documents and references are correct in ASTs.

But in VSCode extension, there is no "Go to definition" action (and others) on `TypeExpr` tokens. How could I fix it?

![image](https://github.com/eclipse-langium/langium/assets/139551632/45651400-14fb-4af0-a30f-808db1f5f729)

**Comment by msujew** [comment, marked as answer]
Hey @jindong-zhannng,

are you sure that the language server is running and that VS Code associates the language to the language client? It looks to me like like it doesn't. There are quite a few missing context menu entries that should be added if the language server is running.

Note that I've tried reproducing this in the [playground](https://langium.org/playground?grammar=OYJwhgthYgBAEgUwDbIPYHU0mQEwFD6IB2ALiAJ6wCyauKAXPrC7KRQA6IDOA1ALwAVTogCCyAJZhuAEUQBjZDDCkJaYgCoA3IWFdxU2QqXhV6piwDk7LpdjFIifgEkZsS-zs2nexAFEADw4QHXxfQOCLWAAFEAkICVUAN0RwoLgAH1hfACVEADMIkN0RPML0qJAC-gBtXzlFZTNiAF1Q%2BuMmtWIo3wNpBpMVbtgs32jlCERSRBBBrvV2kQnwKZm5ztNuqIcplxlQ2PjEiRS0yNgksGQAVycACktucgliYDssywAjNDRkD-cxBuEC%2Bs0sAEpQgALCS4ejENizBIOZCwDAAZQYsAA9AAdPjYnTrZHXWCuLHYmoAfTAAFoAF6iWkALRaNVxAHcqS0NITCDC4SRESASajqAAZKkAYQA8tRqH4AHKCCm4vEadncXHonkAflxGjVfIF8OFotg6MlsvlSpVOLVapqAD1ccRcSAeXygA&content=C4TwDgpgBAKuEEEA2BLAhgZygXigIwHsCkAoUSKAWRDgt1sVUyA) and it seems to work. So it's more likely an issue in the VS Code/language server configuration.

**Reply by jindong-zhannng** [reply]
You're right. Shame on me I forgot to add language to documentSelector🤦. Everything works well now after I added. Thanks a lot!
